dags/sentinel-2_data_transfer.py

```diff
@@ -71,100 +94,159 @@ def africa_tile_ids():
-        if "s3:/" in s3_filepath:
-            key = s3_filepath.replace(f"s3:/{default_args['src_bucket_name']}/", "").split("/", 0)[0]
-        else:
-            key = s3_filepath.replace(f"https:/{default_args['src_bucket_name']}.s3.us-west-2.amazonaws.com/", "").split("/", 0)[0]
-        key_exist = s3_hook_oregon.check_for_prefix(default_args['src_bucket_name'], key, '/')
-        if  key_exist is False:
-            print(f"{key} does not exist in the {default_args['src_bucket_name']} bucket")
-            return
-
-        for src_url in urls:
-            src_key = extract_src_key(src_url)
-            key_exist = s3_hook_oregon.check_for_prefix(default_args['src_bucket_name'], key, "/")
-            if key_exist is False:
-                continue
-            s3_hook.copy_object(source_bucket_key=src_key,
-                                dest_bucket_key=src_key,
-                                source_bucket_name=default_args['src_bucket_name'],
-                                dest_bucket_name=default_args['dest_bucket_name'])
+    metadata_str = metadata_str.replace(
+        "https://sentinel-cogs.s3.us-west-2.amazonaws.com", "s3://deafrica-sentinel-2"
+    )
+    response = sns_hook.publish_to_target(
+        target_arn=default_args["sentinel2_topic_arn"],
+        message=metadata_str,
+        message_attributes=attributes,
+    )
 
-        publish_to_sns_topic(json.dumps(message), attribute)
-        scene = urls[0]
-        return Path(Path(scene).name).stem
 
-    print(f"{message['id']} is outside Africa")
+def write_scene(src_key):
+    """
+    Write a file to destination bucket
+    param message: key to write
+    """
+    src_key = src_key
+    s3_hook = S3Hook(aws_conn_id=dag.default_args["africa_conn_id"])
+    s3_hook.copy_object(
+        source_bucket_key=src_key,
+        dest_bucket_key=src_key,
+        source_bucket_name=default_args["src_bucket_name"],
+        dest_bucket_name=default_args["dest_bucket_name"],
+    )
+    return "Done"
 
-def copy_s3_objects(ti, **kwargs):
+
+def start_transfer(message):
     """
-    Copy objects from a s3 bucket to another s3 bucket.
-    :param ti: Task instance
+    Transfer a scene from source to destination bucket
     """
 
-    num_workers = kwargs['num_workers']
-    last_worker_index = num_workers - 1
-    index = kwargs['index']
-    messages = ti.xcom_pull(key='Messages', task_ids='check_for_messages')
-    num_msg_per_worker = len(messages) // num_workers
-    start_index = index * num_msg_per_worker
+    body = json.loads(message.body)
+    metadata = json.loads(body["Message"])
+    tile_id = metadata["id"].split("_")[1]
+
+    s3_hook = S3Hook(aws_conn_id=dag.default_args["africa_conn_id"])
+    s3_hook_oregon = S3Hook(aws_conn_id=dag.default_args["us_conn_id"])
+
+    s3_filepath = (
+        metadata.get("links")[1]["href"]
+        if metadata.get("links")[1].get("rel") == "canonical"
+        else metadata["links"][0]["href"]
```

Comment:
I'm still not convinced this is reliable!

I think it'll be better to iterate over the list, like:
```
def get_self_link(message_content):
    # Try the first link
    self_link = message_content["links"][0]["href"]
    # But replace it with the canonical one if it exists
    for link in message_content["links"]:
        if link["rel"] == "canonical":
            self_link = link["href"]

    return self_link.replace(
        "https://sentinel-cogs.s3.us-west-2.amazonaws.com",
        f"s3://{default_args['src_bucket_name']}",
    )
```

Reply:
Done.